third_party/2and3/first.pyi

```diff
@@ -0,0 +1,15 @@
+from typing import Any, Callable, Iterable, Optional, overload, TypeVar, Union
+
+_T = TypeVar('_T')
+_S = TypeVar('_S')
+
+@overload
+def first(iterable: Iterable[_T]) -> Optional[_T]: ...
+
+
+@overload
+def first(iterable: Iterable[_T], default: _S) -> Union[_T, _S]: ...
+
+
+@overload
+def first(iterable: Iterable[_T], default: _S, key: Optional[Callable[[_T], Any]]) -> Union[_T, _S]: ...
```

Comment:
You're missing support for the case where `key` is given but `default` is not.

Also, don't put any blank lines between the overloads.